src/sentry/event_manager.py

```diff
@@ -608,18 +608,42 @@ def save(self, project, raw=False):
         # store a reference to the group id to guarantee validation of isolation
         event.data.bind_ref(event)
 
-        try:
-            with transaction.atomic(using=router.db_for_write(EventMapping)):
-                EventMapping.objects.create(project=project, group=group, event_id=event_id)
-        except IntegrityError:
+        # When an event was sampled, the canonical source of truth
+        # is the EventMapping table since we aren't going to be writing out an actual
+        # Event row. Otherwise, if the Event isn't being sampled, we can safely
```

Comment:
I don't think "safely" is the right word to use here since the ID can actually exist in both tables at this point.